package com.proyecto.servicios.client;

import com.proyecto.servicios.config.GestoPagoFeignConfig;
import com.proyecto.servicios.model.gestopago.GestoPagoAuthResponse;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.http.MediaType;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestParam;

@FeignClient(
        name = "gestoPagoAuth",
        url = "${gestopago.auth.url}",
        configuration = GestoPagoFeignConfig.class
)
public interface GestoPagoAuthClient {

    @PostMapping(
            value = "/sistema/app/jwt-gp/authenticate/",
            produces = MediaType.APPLICATION_JSON_VALUE
    )
    GestoPagoAuthResponse authenticate(
            @RequestParam("idDistribuidor") Integer idDistribuidor,
            @RequestParam("codigoDispositivo") String codigoDispositivo,
            @RequestParam("password") String password
    );
}

